API.md

```diff
@@ -101,6 +101,17 @@ Conflict.
   - 409 Conflict (StripeID is already set)
   - 500
 
+### DELETE `/user`
+
+Deletes the user and all of their data.
+
+* Requires valid JWT: `true`
+* Returns:
+  - 204
+  - 401 (missing JWT)
+  - 404 (when there is no such user)
+  - 500 (on any other error)
```

Comment:
nit or follow up: we could probably move this to the top of the API docs right? I assume 401, 404 are the same for all endpoints where applicable, and then 500 is the default error. So at the top we have a default response codes section and then each endpoint simply states if they have a non default response.